src/config.ts

```diff
@@ -51,53 +52,89 @@ export default class ConfigService {
    */
   #currentVersion: string | null = null;
 
-  constructor(path: string) {
-    this.#uri = path;
+  constructor(uri: string) {
+    this.#uri = uri;
   }
 
-  async resolve() {
-    let url: URL | null = null;
+  /**
+   * Fetch and return the configuration module.
+   *
+   * @param deadline When a deadline is provided, fetching the module via HTTP
+   *                 will be aborted if the deadline is exceeded. Specified in
+   *                 milliseconds.
+   */
+  async fetchConfig(deadline?: number) {
```

Comment:
Do we want this to be optional instead of either requiring this or giving a default?

Reply:
I'll make it required, seems reasonable :+1: